table/metadata.go

```diff
@@ -156,6 +166,42 @@ type commonMetadata struct {
 	Refs               map[string]SnapshotRef  `json:"refs"`
 }
 
+func (c *commonMetadata) Equals(other *commonMetadata) bool {
```

Comment:
hmm, shouldn't that get handled in an `__eq__` method in pyiceberg on the metadata objects? Because I'm not seeing that handling in pyiceberg anywhere unless i'm missing something. If this is sufficient for now, then I'm good with leaving this until we have a clear example where it is insufficient

Reply:
For example here: https://github.com/apache/iceberg-python/blob/4282d2f342167255bb65034a12131ce8fe6a9c04/pyiceberg/table/metadata.py#L377-L393

If `schemas` is missing, we'll move the schema in `schema` to `schemas`, and set `current-schema-id`.